tests/framework/test_time.py

```diff
@@ -56,94 +62,106 @@ def step_column(sim):
     return sim._population._population.step_size
 
 
+def take_step(sim):
+    old_time = sim._clock.time
+    sim.step()
+    new_time = sim._clock.time
+    return new_time - old_time
+
+
 class StepModifier(MockGenericComponent):
-    def __init__(self, name, step_modifier):
+    def __init__(self, name, step_modifier_even, step_modifier_odd):
         super().__init__(name)
-        self.step_modifier = step_modifier
+        self.step_modifier_even = step_modifier_even
+        self.step_modifier_odd = step_modifier_odd
+        self.ts_pipeline_value = None
 
     def setup(self, builder) -> None:
         super().setup(builder)
         builder.value.register_value_modifier("simulant_step_size", self.modify_step)
+        self.rate_pipeline = builder.value.register_value_producer(
+            "test_rate",
+            source=lambda idx: pd.Series(1.75, index=idx),
+            preferred_post_processor=rescale_post_processor,
+        )
 
-    def modify_step(self, index):
-        return pd.Series(pd.Timedelta(days=self.step_modifier), index=index)
-
+    def on_time_step(self, event: Event) -> None:
+        self.ts_pipeline_value = self.rate_pipeline(event.index)
 
-def test_align_times(SimulationContext, base_config, components):
+    def modify_step(self, index):
```

Comment:
These tests are getting intensely detailed and I think some brief comments would go a long way